app/src/main/java/com/github/kulebin/myfoursquareapp/imageLoader/ImageLoader.java

```diff
@@ -0,0 +1,136 @@
+package com.github.kulebin.myfoursquareapp.imageLoader;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Handler;
+import android.widget.ImageView;
+
+import com.github.kulebin.myfoursquareapp.http.IHttpClient;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+class ImageLoader implements IImageLoader {
+
+    interface ISizeParams {
+
+        int getWidth();
+
+        int getHeight();
+    }
+
+    private static final int CHUNK_SIZE_POWER = 16; //chunk size = 2^x
+
+    private ISizeParams mSizeParamsCallback;
+
+    private final IHttpClient.IOnResultConvert<Bitmap> mBitmapConverter = new IHttpClient.IOnResultConvert<Bitmap>() {
+
+        @Override
+        public Bitmap convert(final InputStream inputStream) throws IOException {
+            final BitmapFactory.Options options = new BitmapFactory.Options();
+            final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream(inputStream.available());
+            final BufferedInputStream buff = new BufferedInputStream(inputStream, 1 << CHUNK_SIZE_POWER);
+            int byteValue;
+
+            while ((byteValue = buff.read()) != -1) {
+                byteArrayOutputStream.write(byteValue);
+            }
+
+            final byte[] byteArray = byteArrayOutputStream.toByteArray();
+            options.inJustDecodeBounds = true;
+            BitmapFactory.decodeByteArray(byteArray, 0, byteArray.length, options);
+            options.inSampleSize = calculateInSampleSize(options, mSizeParamsCallback.getWidth(), mSizeParamsCallback.getHeight());
+            options.inJustDecodeBounds = false;
+
+            return BitmapFactory.decodeByteArray(byteArray, 0, byteArray.length, options);
+        }
+    };
+
+    @Override
+    public void draw(final String pUrl, final ImageView pView) {
+        draw(pUrl, pView, pView.getWidth(), pView.getHeight());
+    }
+
+    @Override
+    public void draw(final String pUrl, final ImageView pView, final int pWidth, final int pHeight) {
+        pView.setTag(pUrl);
+        final Handler handler = new Handler();
+        new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                mSizeParamsCallback = initSizeParamsCallback(pWidth, pHeight);
+
+                IHttpClient.Impl.get().doRequest(pUrl, new IHttpClient.IOnResult<Bitmap>() {
+
+                            @Override
+                            public void onSuccess(final Bitmap pBitmap) {
+                                final Bitmap bitmap = resizeBitmap(pBitmap, pWidth, pHeight);
```

Comment:
I've changed a little bit logic, but in any case I need resize the bitmap twice, in first place to scale image and second one to crop image to it fits the set width and height